src/components/Card.js

```diff
@@ -1,44 +1,84 @@
 export default class Card {
   #name;
   #link;
+  #isLiked;
+  #owner;
+  #cardId;
+  #createdAt;
   #cardTemplateSelector;
   #cardElement;
   #cardLikeButton;
   #cardDeleteButton;
   #cardImageEl;
   #cardTitleEl;
   #handleCardClick;
+  #handleCardLike;
+  #handleCardDislike;
+  #handleCardDelete;
+  #handleClickDeleteButton;
+  #confirmDeletePopup;
 
-  constructor({ name, link }, cardTemplateSelector, handleCardClick) {
+  constructor(
+    { name, link, isLiked, owner, _id, createdAt },
+    cardTemplateSelector,
+    handleCardClick,
+    confirmDeletePopup,
+    handleCardLike,
+    handleCardDislike,
+    handleCardDelete
+  ) {
     this.#name = name;
     this.#link = link;
+    this.#isLiked = isLiked;
+    this.#owner = owner;
+    this.#cardId = _id;
+    this.#createdAt = createdAt;
     this.#cardTemplateSelector = cardTemplateSelector;
     this.#handleCardClick = handleCardClick;
+    this.#handleCardLike = handleCardLike;
+    this.#handleCardDislike = handleCardDislike;
+    this.#handleCardDelete = handleCardDelete;
+    this.#confirmDeletePopup = confirmDeletePopup;
+
+    this.#handleClickDeleteButton = this._handleClickDeleteButton.bind(this);
   }
 
   #setEventListeners() {
-    this.#cardLikeButton = this.#cardElement.querySelector(".card__like-button");
     this.#cardDeleteButton = this.#cardElement.querySelector(".card__delete-button");
 
     this.#cardLikeButton.addEventListener("click", () => {
-      this.#handleLikeIcon();
+      this.#handleClickLikeButton();
     });
+
     this.#cardDeleteButton.addEventListener("click", () => {
-      this.#handleDeleteIcon();
+      this.#confirmDeletePopup.setSubmitAction(this.#handleClickDeleteButton);
+      this.#confirmDeletePopup.open();
     });
 
     this.#cardImageEl.addEventListener("click", () => {
       this.#handleCardClick(this.#link, this.#name);
     });
   }
 
-  #handleLikeIcon() {
-    this.#cardLikeButton.classList.toggle("card__liked-heart");
+  _handleClickDeleteButton() {
+    this.#handleCardDelete(this.#cardId).then(() => {
+      this.#confirmDeletePopup.close();
+      this.#cardElement.remove();
+    });
   }
 
-  #handleDeleteIcon() {
-    this.#cardElement.remove();
-    this.#cardElement = null;
+  #handleClickLikeButton() {
+    if (this.#isLiked) {
+      this.#handleCardDislike(this.#cardId).then(() => {
+        this.#isLiked = !this.#isLiked;
+      });
+      this.#cardLikeButton.classList.remove("card__liked-heart");
+    } else {
+      this.#handleCardLike(this.#cardId).then(() => {
+        this.#isLiked = !this.#isLiked;
+      });
+      this.#cardLikeButton.classList.add("card__liked-heart");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.